src/main/scala/strawman/collection/mutable/BitSet.scala

```diff
@@ -1,27 +1,119 @@
+import scala.Predef.{require}
 
-trait BitSet
+/**
+  * A class for mutable bitsets.
+  *
+  * $bitsetinfo
+  *
+  * @see [[http://docs.scala-lang.org/overviews/collections/concrete-mutable-collection-classes.html#mutable_bitsets "Scala's Collection Library overview"]]
+  * section on `Mutable Bitsets` for more information.
+  *
+  * @define Coll `BitSet`
+  * @define coll bitset
+  * @define orderDependent
+  * @define orderDependentFold
+  * @define mayNotTerminateInf
+  * @define willNotTerminateInf
+  */
+@SerialVersionUID(8483111450368547763L)
+class BitSet(protected[collection] final var elems: Array[Long])
   extends collection.BitSet
     with collection.BitSetLike[BitSet]
-    with SortedSet[Int] {
+    with SortedSet[Int]
+    with SetMonoTransforms[Int, BitSet] // Override monomorphic transformations to return a BitSet rather than a SortedSet[Int]
+    with Serializable {
 
-  private[collection] def elems: Array[Long]
+  def this(initSize: Int) = this(new Array[Long](math.max((initSize + 63) >> 6, 1)))
+
+  def this() = this(0)
+
+  def fromIterable[B](coll: collection.Iterable[B]): collection.mutable.Set[B] =
+    collection.mutable.Set.fromIterable(coll)
+
+  protected def constrainedFromIterable[B : Ordering](it: collection.Iterable[B]): collection.mutable.SortedSet[B] =
+    collection.mutable.SortedSet.constrainedFromIterable(it)
+
+  protected[this] def fromIterableWithSameElemType(coll: collection.Iterable[Int]): BitSet =
+    BitSet.fromIterable(coll)
+
+  protected[collection] final def nwords: Int = elems.length
+
+  protected[collection] final def word(idx: Int): Long =
+    if (idx < nwords) elems(idx) else 0L
+
+  protected[collection] def fromBitMaskNoCopy(elems: Array[Long]): BitSet = new BitSet(elems)
+
+  def addInPlace(elem: Int): this.type = {
+    require(elem >= 0)
+    if (!contains(elem)) {
+      val idx = elem >> LogWL
+      updateWord(idx, word(idx) | (1L << elem))
+    }
+    this
+  }
+
+  def removeInPlace(elem: Int): this.type = {
+    require(elem >= 0)
+    if (contains(elem)) {
+      val idx = elem >> LogWL
+      updateWord(idx, word(idx) & ~(1L << elem))
+    }
+    this
+  }
+
+  def clear(): Unit = {
+    elems = new Array[Long](elems.length)
+  }
+
+  protected final def updateWord(idx: Int, w: Long): Unit = {
+    ensureCapacity(idx)
+    elems(idx) = w
+  }
+
+  protected final def ensureCapacity(idx: Int): Unit = {
+    require(idx < MaxSize)
+    if (idx >= nwords) {
+      var newlen = nwords
+      while (idx >= newlen) newlen = math.min(newlen * 2, MaxSize)
+      val elems1 = new Array[Long](newlen)
+      Array.copy(elems, 0, elems1, 0, nwords)
+      elems = elems1
+    }
+  }
+
+  def get(elem: Int): Option[Int] = scala.Predef.???
+
+  def unconstrained: collection.Set[Int] = this
+
+  def empty: BitSet = BitSet.empty
+
+  override def clone(): BitSet = {
+    val elems1 = new Array[Long](elems.length)
+    Array.copy(elems, 0, elems1, 0, elems.length)
```

Comment:
Why are we doing it this way instead of `java.util.Arrays.copyOf(elems, elems.length)` or an equivalent?  The `copyOf` approach has a better chance of eliding pointless array initialization, and better captures the intent (i.e. make a copy with the same content).